docs/4. repository/2. file-repositories.mdx

```diff
@@ -0,0 +1,135 @@
+
+## Table of Contents
+
+- [RepositoryV1 vs RepositoryV2](#repositoryv1-vs-repositoryv2)
+  - [RepositoryV1](#repositoryv1)
+  - [RepositoryV2](#repositoryv2)
+- [Download](#download)
+
+
+## RepositoryV1 vs RepositoryV2
+
+### RepositoryV1
+RepositoryV1 is the legacy storage backend for files. It is simply a file system based on how file paths are given in the [Solidity metadata](/docs/metadata). file.
+
+An [example metadata](https://repo.sourcify.dev/contracts/full_match/1/0x801f3983c7baBF5E6ae192c84E1257844aDb4b4D/metadata.json) source file paths look like this for the "full_match" contract `0x801f3983c7baBF5E6ae192c84E1257844aDb4b4D` on Ethereum Mainnet (1):
+```json
+{
+  "sources": {
+    "erc20/IERC20.sol": {
+      "keccak256": "0xa38ec4e151e4d397d05bdfb94e6e4eb91e57a9fca3bc1c655289a4adf31a58fa",
+      "license": "MIT",
+      "urls": [
+        "bzz-raw://312e850e36efbf0f2450896c213b23dc0a28150e051bcbf933a8b9211627c44b",
+        "dweb:/ipfs/QmWsyisPjDwTJrTMhsGZa4JHiCS63mWfsyVQKbaijWGdmK"
+      ]
+    },
+    "erc20/airdrop.sol": {
+      "keccak256": "0xea27a3e2c4179a064caf9fe9a198addd526fd1d1ea467ea474a0c069e6eac957",
+      "urls": [
+        "bzz-raw://6a86bc69b99876768bdbddba504410cf60b33681e1203a36d98840bf2ab8a42b",
+        "dweb:/ipfs/QmRZSqNfAPduoPoUJ6BM4NpBTbTKBqg5Mz5YBNpaUz4TfQ"
+      ]
+    }
+  },
+}
+```
+
+These files will be like below ([see in repo](https://repo.sourcify.dev/contracts/full_match/1/0x801f3983c7baBF5E6ae192c84E1257844aDb4b4D/)):
+```
+contracts/full_match/1/0x801f3983c7baBF5E6ae192c84E1257844aDb4b4D/
+├── metadata.json
+└── sources/
+    └── erc20/
+        ├── IERC20.sol
+        └── airdrop.sol
+```
+The problem with this is the part `"erc20/airdrop.sol"` is not necessarily a valid file path but a ["source unit name"](https://docs.soliditylang.org/en/v0.8.27/path-resolution.html#virtual-filesystem:~:text=assigned%20a%20unique-,source%20unit%20name,-which%20is%20an) in Solidity, i.e. arbitrary strings. This may cause issues on file systems as well as when pinning to IPFS.
+
+### RepositoryV2
+
+RepositoryV2 is the format where we normalize the file names with their keccak256 hashes (source files must have a `keccak256` field in the metadata). So the example above would look like this:
+
+```
+contracts/full_match/1/0x801f3983c7baBF5E6ae192c84E1257844aDb4b4D/
+├── metadata.json
+└── sources/
+    └── 0xa38ec4e151e4d397d05bdfb94e6e4eb91e57a9fca3bc1c655289a4adf31a58fa.sol
+    └── 0xea27a3e2c4179a064caf9fe9a198addd526fd1d1ea467ea474a0c069e6eac957.sol
+```
+
+The files are exactly the same so their IPFS hashes will not change, and you can look up the metadata file to find the original path-alike source unit names.
+
+## IPFS
+
+Unfortunatelly publishing under IPNS is temporarily disabled. This is because of the difficulty of managing the whole filesystem over IPFS (with MFS etc.) and updating the IPNS regularly. 
```

Comment:
Does it make sense to cite this? Probably our users don't even know that there was an IPNS repository, let's not confuse them

Reply:
I'd say absolutely yes. A lot of people ask us about the IPNS thing